code/modules/reagents/reagent_containers/food/drinks.dm

```diff
@@ -263,11 +263,17 @@ If you add a drink with an empty icon sprite, ensure it is in the same folder, e
 		AddOverlays(filling)
 
 //heehoo bottle flipping
-/obj/item/reagent_containers/food/drinks/waterbottle/throw_impact()
+/obj/item/reagent_containers/food/drinks/waterbottle/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
 	. = ..()
 	if(!QDELETED(src))
-		if(prob(10)) // landed upright in some way
-			if(prob(10)) // landed upright on ITS CAP (1% chance)
+		var/is_psi_apex_or_higher = FALSE
+		var/mob/living/thrower = throwingdatum.thrower?.resolve()
+		if(istype(thrower))
+			is_psi_apex_or_higher = thrower.psi && thrower.psi.get_rank() >= PSI_RANK_APEX
+		var/upright_chance = is_psi_apex_or_higher ? 100 : 10
+		var/cap_chance = is_psi_apex_or_higher ? 50 : 10
+		if(prob(upright_chance)) // landed upright in some way
+			if(prob(cap_chance)) // landed upright on ITS CAP (1% chance)
```

Comment:
this needs to be commented on what the hell is going on here

I don't want to say this is shitcode but in 1 or 2 years someone will look at this and ask "why the fuck is a waterbottle throw proc checking for psionics"